06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.allIndex = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+    this.validatedFile = {};
+  }
+
+  /**
+   * Validate Method
+   * @param {Object} validateFile an object to be validated
+   * @returns {Object} file that has been validated
+   */
+  validateFileFunc(validateFile) {
+    this.validatedFile = validateFile;
+    if (validateFile.title !== undefined && validateFile.text !== undefined) {
+      return validateFile;
+    } return false;
+  }
+
+  /**
+   * Sanitizer
+   * @param {Object} BookObject recieves json object
+   * @return {Object} returns an Object of names
```

Comment:
Looking at the implementation of this method, it doesn't return as described by the JSDoc.